### PR TITLE
[TIMOB-26216] iOS: Force clean build if deployType changes

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-26216

### CODE REVIEW

File: iphone/cli/commands/_build.js [hunk 1]
```diff
@@ -2692,6 +2692,14 @@ iOSBuilder.prototype.checkIfNeedToRecompile = function checkIfNeedToRecompile()
 			return true;
 		}
 
+		// check if the deployType flag has changed
+		if (this.deployType !== manifest.deployType) {
+			this.logger.info(__('Forcing rebuild: deployType flag changed since last build'));
```
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: Technically, deploy type is not a flag. :)

**Review by cb1kenobi — APPROVED** [review]

**Review by cb1kenobi — APPROVED** [review]

**Comment by craigrouse** [comment]
@hansemannn I'm experiencing this issue now (and it's my first time attempting to build a Titanium app). I'm using Appcelerator Studio. Is there a workaround for this problem that I can use right now?

**Comment by hansemannn** [comment]
@craigrouse You mean that your "test" builds are unencrypted? You simply need to clean the build directory before switchtng from a "development" to a "test" target. All this PR does is to clear the build directory if the tooling notices a change in the deploy-type. 

Alternatively, you can patch your SDK manually by replacing the lines of code of this PR with your stable SDK (located in `~/Users/hknoechel/Library/Application Support/Titanium/mobilesdk/osx/7.2.0.GA/iphone/cli/commands.

**Comment by craigrouse** [comment]
@hansemannn d'oh! I commented on the wrong PR 🤦‍♂️ . My comment should have been on this one: https://github.com/appcelerator/titanium_mobile/pull/10123. I'll move it. Thanks for the quick reply.

**Pull request merged**